heartbeat/monitors/factory.go

```diff
@@ -179,7 +195,15 @@ func newCommonPublishConfigs(info beat.Info, cfg *conf.C) (pipetool.ConfigEditor
 	}
 
 	// Early stage processors for setting data_stream, event.dataset, and index to write to
-	preProcs, err := preProcessors(info, settings, sf.Type)
+
+	// Use the monitor-specific location if possible, otherwise use the beat's location
```

Comment:
This caught me off-guard, can we add comment about Fleet using monitor specific location whereas as a User I can use top level to avoid duplicating it over all places.

Reply:
good point, will push that comment shortly